File: Task3/ex3Swift/palindrom.swift
import Foundation

/// Проверяет, является ли число трехзначным палиндромом
/// - Parameter num: Проверяемое число
/// - Returns: true, если число является трехзначным палиндромом
func isThreeDigitPalindrome(_ num: Int) -> Bool {
    let absNum = abs(num)
    guard absNum >= 100 && absNum <= 999 else { return false }
    return (absNum / 100) == (absNum % 10)
}

func main() {
    print("Введите числа через пробел:", terminator: " ")
    guard let input = readLine() else { return }
    
    let numbers = input.components(separatedBy: .whitespaces)
    var result = "Результат: "
    var isFirst = true
    
    for numStr in numbers {
        guard let num = Int(numStr) else { continue }
        
        if abs(num) >= 100 && abs(num) <= 999 {
            if !isFirst {
                result += ", "
            }
            result += "\(num) - \(isThreeDigitPalindrome(num) ? "YES" : "NO")"
            isFirst = false
        }
    }
    
    print(result)
}

// Запуск программы
main()

File: Task3/ex3TypeScript/src/palindrom.ts
/**
 * Проверяет, является ли число трёхзначным палиндромом.
 * @param num - Проверяемое число
 * @returns true, если число трёхзначное и является палиндромом
 */
function isThreeDigitPalindrome(num: number): boolean {
    const absNum = Math.abs(num); // Работаем с модулем числа
    if (absNum < 100 || absNum > 999) return false; // Проверка на трёхзначность
    return Math.floor(absNum / 100) === absNum % 10; // Сравниваем первую и последнюю цифры
}

function main(): void {
    // Настройка ввода/вывода
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    // Запрос ввода и обработка данных
    rl.question('Введите числа через пробел: ', (input: string) => {
        const numbers = input.trim().split(/\s+/); // Разбиваем строку на числа
        let result = 'Результат: ';
        let isFirstOutput = true; // Флаг для форматирования вывода

        // Обработка каждого числа
        for (const numStr of numbers) {
            const num = parseInt(numStr, 10);
            
            // Пропускаем некорректные значения
            if (isNaN(num)) continue;
            
            // Проверяем трёхзначные числа
            if (Math.abs(num) >= 100 && Math.abs(num) <= 999) {
                if (!isFirstOutput) result += ', '; // Добавляем разделитель
                result += `${num} - ${isThreeDigitPalindrome(num) ? 'YES' : 'NO'}`;
                isFirstOutput = false;
            }
        }

        console.log(result); // Вывод результата
        rl.close(); // Закрываем ввод
    });
}

// Запуск программы
main();